Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Internal/FbOptions.cs

```diff
@@ -15,16 +15,28 @@
 
 //$Authors = Jiri Cincura (jiri@cincura.net), Jean Ressouche, Rafael Almeida (ralms@ralms.net)
 
+using System;
+using System.Linq;
 using FirebirdSql.EntityFrameworkCore.Firebird.Infrastructure.Internal;
+using FirebirdSql.EntityFrameworkCore.Firebird.Utilities;
 using Microsoft.EntityFrameworkCore.Infrastructure;
 
 namespace FirebirdSql.EntityFrameworkCore.Firebird.Internal
 {
 	public class FbOptions : IFbOptions
 	{
+
+		private Lazy<FbSettings> _fbSettings;
+		public virtual FbSettings Settings => _fbSettings.Value;
+
 		public void Initialize(IDbContextOptions options)
 		{
 			var fbOptions = GetOptions(options);
+
+			_fbSettings = new Lazy<FbSettings>(() => fbOptions.Connection != null ?
```

Comment:
Formatting.

Reply:
Would it look better?

public void Initialize(IDbContextOptions options)
{
    var fbOptions = GetOptions(options); 
    _fbSettings = new Lazy<FbSettings>(() => fbOptions.Connection != null
    ? new FbSettings().GetSettings(fbOptions.Connection)
    : new FbSettings().GetSettings(fbOptions.ConnectionString)); 
}

Reply:
Yes.

Reply:
Also your `fbOptions` can be `null` and you should probably handle that.

Reply:
I will correct the formatting!